### PR TITLE
planner: support trace column prune

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/29661

### What is changed and how it works?

support trace column prune

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/rule_column_pruning.go [hunk 1]
```diff
@@ -63,32 +65,38 @@ func exprHasSetVarOrSleep(expr expression.Expression) bool {
 
 // PruneColumns implements LogicalPlan interface.
 // If any expression has SetVar function or Sleep function, we do not prune it.
-func (p *LogicalProjection) PruneColumns(parentUsedCols []*expression.Column) error {
+func (p *LogicalProjection) PruneColumns(parentUsedCols []*expression.Column, opt *logicalOptimizeOp) error {
 	child := p.children[0]
 	used := expression.GetUsedList(parentUsedCols, p.schema)
+	prunedColumns := make([]*expression.Column, 0)
```
Comment (JmPotato) [inline planner/core/rule_column_pruning.go:71]: I guess it should be like this?

```suggestion
	prunedColumns := make([]*expression.Column, 0, 0)
```

File: planner/core/rule_column_pruning.go [hunk 2]
```diff
@@ -166,10 +180,13 @@ func (la *LogicalAggregation) PruneColumns(parentUsedCols []*expression.Column)
 	return nil
 }
 
-func pruneByItems(old []*util.ByItems) (new []*util.ByItems, parentUsedCols []*expression.Column) {
+func pruneByItems(p LogicalPlan, old []*util.ByItems, opt *logicalOptimizeOp) (new []*util.ByItems,
+	parentUsedCols []*expression.Column) {
+	prunedByItems := make([]*util.ByItems, 0)
```
Comment (JmPotato) [inline planner/core/rule_column_pruning.go:185]: Ditto.

**Review by rebelice — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**